src/datasets/table.py

```diff
@@ -0,0 +1,591 @@
+
+    @wraps_arrow_table_method(pa.Table.flatten)
+    def flatten(self, *args, **kwargs):
+        replay = [("flatten", copy.deepcopy(args), copy.deepcopy(kwargs))]
+        replay = self._merge_replay(replay)
+        return MemoryMappedTable(self.table.flatten(*args, **kwargs), self.path, replay)
+
+    @wraps_arrow_table_method(pa.Table.combine_chunks)
+    def combine_chunks(self, *args, **kwargs):
+        replay = [("combine_chunks", copy.deepcopy(args), copy.deepcopy(kwargs))]
+        replay = self._merge_replay(replay)
+        return MemoryMappedTable(self.table.combine_chunks(*args, **kwargs), self.path, replay)
+
+    @wraps_arrow_table_method(pa.Table.cast)
+    def cast(self, *args, **kwargs):
+        replay = [("cast", copy.deepcopy(args), copy.deepcopy(kwargs))]
+        replay = self._merge_replay(replay)
+        return MemoryMappedTable(self.table.cast(*args, **kwargs), self.path, replay)
+
+    @wraps_arrow_table_method(pa.Table.add_column)
+    def add_column(self, *args, **kwargs):
+        replay = [("add_column", copy.deepcopy(args), copy.deepcopy(kwargs))]
+        replay = self._merge_replay(replay)
+        return MemoryMappedTable(self.table.add_column(*args, **kwargs), self.path, replay)
+
+    @wraps_arrow_table_method(pa.Table.append_column)
+    def append_column(self, *args, **kwargs):
+        replay = [("append_column", copy.deepcopy(args), copy.deepcopy(kwargs))]
+        replay = self._merge_replay(replay)
+        return MemoryMappedTable(self.table.append_column(*args, **kwargs), self.path, replay)
+
+    @wraps_arrow_table_method(pa.Table.remove_column)
+    def remove_column(self, *args, **kwargs):
+        replay = [("remove_column", copy.deepcopy(args), copy.deepcopy(kwargs))]
+        replay = self._merge_replay(replay)
+        return MemoryMappedTable(self.table.remove_column(*args, **kwargs), self.path, replay)
+
+    @wraps_arrow_table_method(pa.Table.set_column)
+    def set_column(self, *args, **kwargs):
+        replay = [("set_column", copy.deepcopy(args), copy.deepcopy(kwargs))]
+        replay = self._merge_replay(replay)
+        return MemoryMappedTable(self.table.set_column(*args, **kwargs), self.path, replay)
+
+    @wraps_arrow_table_method(pa.Table.rename_columns)
+    def rename_columns(self, *args, **kwargs):
+        replay = [("rename_columns", copy.deepcopy(args), copy.deepcopy(kwargs))]
+        replay = self._merge_replay(replay)
+        return MemoryMappedTable(self.table.rename_columns(*args, **kwargs), self.path, replay)
+
+    @wraps_arrow_table_method(pa.Table.drop)
+    def drop(self, *args, **kwargs):
+        replay = [("drop", copy.deepcopy(args), copy.deepcopy(kwargs))]
+        replay = self._merge_replay(replay)
+        return MemoryMappedTable(self.table.drop(*args, **kwargs), self.path, replay)
+
+
+BlockPart = TypeVar("BlockPart", TableBlock, List[TableBlock], List[List[TableBlock]])
```

Comment:
why would you need to handle lists of lists?

Reply:
First axis concatenates the rows, and second axis concatenates the columns.
This way we can implement Dataset.add_column for example.

Reply:
Oh ok, I had missed the fact that concatenation could be "horizontal" too. worth a comment?

Reply:
yup I'm adding one